maven-core/src/test/java/org/apache/maven/project/ProjectBuilderTest.java

```diff
@@ -84,4 +87,22 @@ public void testVersionlessManagedDependency()
             // this is expected
         }
     }
+
+    public void testResolveDependencies() throws Exception {
+        File pomFile = new File( "src/test/resources/projects/basic-resolveDependencies.xml" );
+        MavenSession mavenSession = createMavenSession( null );
+        ProjectBuildingRequest configuration = new DefaultProjectBuildingRequest();
+        configuration.setRepositorySession( mavenSession.getRepositorySession() );
+        configuration.setResolveDependencies(true);
```

Comment:
Can we add a test in which this is false and assert that resolution actually does not happen ?